Automattic/jetpack
# Blog Stats Block: Include Visitors Data

Fixes #11428 

## Proposed changes:

* Adds the number of visitors to a site as something that can be displayed as part of the Blog Stats block.

## Jetpack product discussion


See original issue (technically relates to the Legacy Widget, but I think it's reasonable for the block to supersede that).

## Does this pull request change what data or activity we track or use?


We already use this data elsewhere, but it makes it visible in a new way.

## Testing instructions:

Enable Beta blocks and add the "Blog Stats" widget to your site. You should now see two radios (an option dropdown felt very overkill for just two options, so I think a radio is most appropriate here). The new one should enable you to display your site visitors instead of views. 

**Note:** we don't collect data of unique visitors for individual posts. You should be prevented from selecting the "Visitors" and "This individual post" combination.

<img width="1529" alt="Screenshot 2024-02-02 at 22 58 58" src="https://github.com/Automattic/jetpack/assets/43215253/6491853c-75a7-424d-b519-3f877c718c53">

cc @kangzj - sorry to give you something else to review :) for what it's worth, I think this is the only significant issue for the Legacy Widget in the backlog, so would be nice to fix it with this new block!

@jeherve - I'd appreciate your advice here regarding what you said in https://github.com/Automattic/jetpack/issues/11428#issuecomment-538898604:

> I don't think that data is available in the WordPress.com REST API right now, so we'd have to extend the API first.

My understanding is that `visitors` in the API is just that, unique visitors. WordPress.com uses that term precisely from launch [here](https://wordpress.com/blog/2012/12/17/unique-visitors/) and refer to it still [here](https://wordpress.com/support/stats/understand-your-sites-traffic/#views-and-visitors). As such, I believe that the comment in https://github.com/Automattic/jetpack/issues/11428#issuecomment-538727734 was mistaken, but just wanted to make sure.
